state/applicationoffers.go

```diff
@@ -243,13 +243,23 @@ func (op *RemoveOfferOperation) internalRemove(offer *crossmodel.ApplicationOffe
 		Id:     offer.ApplicationName,
 		Assert: bson.D{{"relationcount", app.doc.RelationCount}},
 	}}
+	var remoteApps []*RemoteApplication
 	for _, rel := range rels {
 		remoteApp, isCrossModel, err := rel.RemoteApplication()
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		if isCrossModel && !op.Force {
-			return nil, jujutxn.ErrTransientFailure
+		if isCrossModel {
```

Comment:
We don't need this logic refactoring here - can't the app removal just be done below with the unit cleanup?

Reply:
It can, but I wanted the ability to pass the force flag through to the destroy remote application operation. As discussed, this would only work when `op.Force == true`, so your point is valid. Have refactored.